apps/eo_web/src/components/ThankYou.tsx

```diff
@@ -0,0 +1,94 @@
+import { type HTMLAttributes } from "react";
+import { Typography } from "@eo/ui";
+import { AllDonePanel } from "./AllDonePanel";
+import { useMutation } from "@tanstack/react-query";
+import axios from "axios";
+import { useNavigate, useSearchParams } from "react-router-dom";
+import { toast } from "react-toastify";
+import { useSurveyStore } from "~/stores/useSurveyStore";
+import { Loading } from "~/components/Loading";
+
+import { useMount } from "~/hooks/useMount";
+import { useState } from "react";
+import { useProfilingStore } from "~/stores/useProfilingStore";
+
+type ThankYouProps = HTMLAttributes<HTMLElement> & {
+  mutationKey: string;
+  mutationFunction: (data: object) => Promise<any>;
+  exitRoute?: string;
+  isProfiling?: boolean;
+}
+
+export const ThankYou = ({ children, mutationKey, mutationFunction, exitRoute = '/', isProfiling = false }: ThankYouProps) => {
+  const [isLoading, setIsLoading] = useState(true);
+
+  const [searchParams] = useSearchParams();
+
+  const { email, phase } = useSurveyStore();
+  const { account, usePayment } = useProfilingStore();
+
+  const submission_id = searchParams.get("submission_id") ?? "";
+
+  const navigate = useNavigate();
+
+  if (!submission_id) {
+    navigate(exitRoute);
+  }
```

Comment:
Another approach to early exit & navigate

```suggestion
  if (!submission_id) {
    return <Navigate to={exitRoute} />
  }
```